### PR TITLE
small fixes, additions to rulebuilder

### PR DESCRIPTION
**Pull request by moesterheld**
## Description
- sort function names
- add implicit to_string fragments
- fix string parameter and primary parameter handling in fragments

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/db/migrations/V20230613154400_AddImplicitToStringFragments.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        .build())
+                .fragmentOutputVariable("gl2_fragment_substring_results")
+                .build();
+    }
+
+    private static RuleFragment createDateFragment() {
+        return RuleFragment.builder()
+                .fragment("""
+                        let gl2_fragment_date_results = parse_date(
+                          value: to_string(),
```
Comment (kodjo-anipah) [inline graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/db/migrations/V20230613154400_AddImplicitToStringFragments.java:97]: to_string(${value}),

Comment (moesterheld) [reply graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/db/migrations/V20230613154400_AddImplicitToStringFragments.java]: good catch

**Review by kodjo-anipah — CHANGES_REQUESTED** [review]
There seems to be an problem with output variables couldn't test the date fragment. 
Also we should deactivate Rulebuilder for FormatDate.java

**Comment by moesterheld** [comment]
> There seems to be an problem with output variables couldn't test the date fragment.

Problem is fixed

> Also we should deactivate Rulebuilder for FormatDate.java

I would have liked to have this available to use `get_date` to parse the date and then use `format_date` to create a string with another format. But I think we have to talk about how to handle dates in the UI in the next cycle as the date won't be available for setting in the form currently. I will deactivate it for now.

**Review by kodjo-anipah — APPROVED** [review]
tested all functions and conditions , LGTM

**Pull request merged**